src/components/GlobalPreferences/Notifications/NotificationsVerify.js

```diff
@@ -0,0 +1,185 @@
+import React, { useEffect, useCallback, useState } from 'react'
+import PropTypes from 'prop-types'
+import { ButtonText, LoadingRing } from '@aragon/ui'
+import { verifyEmailToken } from './notification-service-api'
+import { VERIFY_SUBSECTION, ExpiredTokenError } from './constants'
+import NotificationsInfoBox, {
+  ICON_SUCCESS,
+  ICON_ERROR,
+  ICON_NEUTRAL,
+} from './NotificationsInfoBox'
+
+export function NotificationsVerify({
+  subsection,
+  onTokenChange,
+  onEmailChange,
+  navigateToNotifications,
+}) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [verified, setVerified] = useState(false)
+  const [error, setError] = useState(null)
+
+  const handleResetAccount = useCallback(() => {
+    // In error states we can clear email and token to reset the state
+    onEmailChange(null)
+    onTokenChange(null)
+    navigateToNotifications()
+  }, [onEmailChange, onTokenChange, navigateToNotifications])
+
+  useEffect(() => {
+    let cancelled = false
+    // Parse token from subsection /verify/[TOKEN] -> [TOKEN]
+    const token = subsection.substring(VERIFY_SUBSECTION.length)
+    verifyEmailToken(token)
+      .then(longLivedToken => {
+        if (!cancelled) {
+          setVerified(true)
+          setIsFetching(false)
+          onTokenChange(longLivedToken)
+        }
+        return longLivedToken
+      })
+      .catch(e => {
+        if (!cancelled) {
+          setError(e)
+          setIsFetching(false)
+          setVerified(false)
+        }
+      })
+    return () => {
+      cancelled = true
+    }
+  }, [subsection, onTokenChange, onEmailChange])
+
+  if (isFetching) {
```

Comment:
This renders an odd looking box:

![image](https://user-images.githubusercontent.com/3072458/64102983-5d30f980-cd71-11e9-8763-2f72cc017380.png)

Reply:
@dizzypaty What do you think we should render for this loading/fetching/async-inbetween state?

Reply:
<img width="761" alt="Screen Shot 2019-09-02 at 12 22 53 pm" src="https://user-images.githubusercontent.com/1992255/64108147-6a9fb100-cd7c-11e9-8a57-b4ade6e48b76.png">

Fixed by centering